scripts/apps/workspace/content/views/FormattingOptionsMultiSelect.tsx

```diff
@@ -1,48 +1,30 @@
 import React from 'react';
 import {TreeSelect} from 'superdesk-ui-framework/react';
-import {IArticleField} from 'superdesk-api';
+import {RICH_FORMATTING_OPTION} from 'superdesk-api';
 import {gettext} from 'core/utils';
-import {
-    getEditor3RichTextFormattingOptions,
-    getEditor3PlainTextFormattingOptions,
-    HAS_RICH_FORMATTING_OPTIONS,
-} from '../components/get-content-profiles-form-config';
 
 interface IProps {
     value: Array<string> | null;
-    fieldId: string;
-    fields: Dictionary<string, IArticleField>;
-    onChange(value: Array<string>, fieldId: string): void;
+    onChange(value: Array<string>): void;
+    options: Array<{value: [RICH_FORMATTING_OPTION, string]}>;
 }
 
 export class FormattingOptionsTreeSelect extends React.Component<IProps> {
     render(): React.ReactNode {
-        const {fields, fieldId} = this.props;
-
-        const isCustomPlainTextField = typeof fields[fieldId] === 'object' && fields[fieldId].field_type === 'text';
-        const isRichOrCustomTextField =
-            Object.keys(HAS_RICH_FORMATTING_OPTIONS).includes(fieldId) || isCustomPlainTextField;
-        const formattingOptions = Object.entries(
-            isRichOrCustomTextField
-                ? getEditor3RichTextFormattingOptions()
-                : getEditor3PlainTextFormattingOptions(),
-        )
-            .map(([notTranslatedOption, translatedOption]) => ({value: [notTranslatedOption, translatedOption]}));
-
-        const values = this.props.value != null
-            ? formattingOptions
-                .filter((option) => this.props.value.includes(option.value[0]))
-                .map((option) => option.value)
-            : [];
+        const lookup = new Map(
+            this.props.options.map(({value}) => value) as unknown as Array<[string, RICH_FORMATTING_OPTION]>,
+        );
+        const values: Array<[string, string]> = (this.props.value ?? []).map((id) => [id, lookup.get(id)]);
```

Comment:
Are you sure type of `values` is correct here? Might be that it's `Array<[RICH_FORMATTING_OPTION, string]>`